### DISCUSSION TITLE
Run always fails with 404 on updating a gist

### DISCUSSION THREAD

**Discussion by PerchunPak** [lowlighter/metrics#1227] [Q&A] [answered]
I recently setuped push to gist, instead of main repo. However, it's now failing with 404 while pushing the changes.

Firstly I thought that I didn't set correct gist ID, but `GET` request to API with the same ID returns expected answer.

[Logs from workflow](https://github.com/PerchunPak/PerchunPak/actions/runs/3037421568), [gist in UI](https://gist.github.com/PerchunPak/c825bee339857f064a9f1829674bae1e), [request URL](https://api.github.com/gists/c825bee339857f064a9f1829674bae1e).

**Comment by lowlighter** [comment, marked as answer]
Hello 👋 !

Yeah it's kind of strange indeed 🤔 
The link used seems to be the correct one since it works (at least using a GET request)

Just to be rule out one possibility first, when you changed your workflow, did you also update your personal access token to grant `gists` scope ? If not it could be that

**Reply by PerchunPak** [reply]
Yeah, sorry. I maybe forgot to click `Save` button in settings.

**Reply by lowlighter** [reply]
No worries, glad it worked 👍
Pretty strange that GitHub chose to send back 404 rather than 403

**Reply by spenserblack** [reply]
>Pretty strange that GitHub chose to send back 404 rather than 403

I'm just guessing, but since gists can be "secret," maybe GitHub wants to pretend that the resource doesn't exist when you don't have access to it.

**Reply by PerchunPak** [reply]
This gist is public.

**Reply by spenserblack** [reply]
Could just be a catch-all rule they have. `if has_access: normal_response; else: 404`.
